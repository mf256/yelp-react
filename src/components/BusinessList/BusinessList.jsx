import React from 'react';
import { PropTypes } from 'prop-types';
import './BusinessList.css';
import Business, { businessType } from '../Business/Business';

class BusinessList extends React.PureComponent {
  render() {
    const { businesses } = this.props;
    return (
      <div className="BusinessList">
        {businesses && businesses.map((b) => (<Business business={b} key={b.id} />))}
      </div>
    );
  }
}

BusinessList.propTypes = {
  businesses: PropTypes.arrayOf(businessType),
};

BusinessList.defaultProps = {
  businesses: [],
};

export default BusinessList;
