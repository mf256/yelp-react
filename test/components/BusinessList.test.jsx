import React from 'react';
import { shallow } from 'enzyme';
import BusinessList from '../../src/components/BusinessList/BusinessList';
import { mockBusinessList } from '../mock/mockData';

describe('BusinessList', () => {
  it('should match snapshot', () => {
    const component = shallow(<BusinessList />);

    expect(component).toMatchSnapshot();
  });

  it('should render items', () => {
    const items = mockBusinessList;
    const hasMore = true;
    const component = shallow(
      <BusinessList
        businesses={items}
        loadMore={jest.fn}
        hasMore={hasMore}
      />,
    );

    expect(component).toMatchSnapshot();
  });

  it('should render last items', () => {
    const items = mockBusinessList;
    const hasMore = false;
    const component = shallow(
      <BusinessList
        businesses={items}
        loadMore={jest.fn}
        hasMore={hasMore}
      />,
    );

    expect(component).toMatchSnapshot();
  });
});
