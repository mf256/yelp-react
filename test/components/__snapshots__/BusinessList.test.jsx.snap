// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BusinessList should match snapshot 1`] = `
<InfiniteScroll
  element="div"
  getScrollParent={null}
  hasMore={true}
  initialLoad={true}
  isReverse={false}
  loadMore={[Function]}
  loader={<Loader />}
  pageStart={0}
  ref={null}
  threshold={250}
  useCapture={false}
  useWindow={true}
>
  <styled.div />
</InfiniteScroll>
`;

exports[`BusinessList should render items 1`] = `
<InfiniteScroll
  element="div"
  getScrollParent={null}
  hasMore={true}
  initialLoad={true}
  isReverse={false}
  loadMore={[Function]}
  loader={<Loader />}
  pageStart={0}
  ref={null}
  threshold={250}
  useCapture={false}
  useWindow={true}
>
  <styled.div>
    <Business
      business={
        Object {
          "address": "1 Test Address",
          "category": "Test Category",
          "city": "Test City",
          "id": 1,
          "imageSrc": "https://test.com/test.jpg",
          "name": "Test Name",
          "rating": 4.5,
          "reviewCount": 90,
          "state": "Test State",
          "zipCode": "10102",
        }
      }
      key="1"
    />
  </styled.div>
</InfiniteScroll>
`;

exports[`BusinessList should render last items 1`] = `
<InfiniteScroll
  element="div"
  getScrollParent={null}
  hasMore={false}
  initialLoad={true}
  isReverse={false}
  loadMore={[Function]}
  loader={<Loader />}
  pageStart={0}
  ref={null}
  threshold={250}
  useCapture={false}
  useWindow={true}
>
  <styled.div>
    <Business
      business={
        Object {
          "address": "1 Test Address",
          "category": "Test Category",
          "city": "Test City",
          "id": 1,
          "imageSrc": "https://test.com/test.jpg",
          "name": "Test Name",
          "rating": 4.5,
          "reviewCount": 90,
          "state": "Test State",
          "zipCode": "10102",
        }
      }
      key="1"
    />
  </styled.div>
</InfiniteScroll>
`;
